### ISSUE TITLE
Re-arrange C++ classes and name spaces

### ISSUE THREAD

**Issue by driverpt** [CorsixTH/CorsixTH#1808]
e.g.: 
 - `Src/gfx/sdl/render_target.h`
 - `Src/font.h`
 - `Src/font/bitmap_font.h`

**Comment by Alberth289346** [comment]
I think the main reason for messiness is lack of a description of naming convention of the files.

Otherwise, C++ isn't Java, having closely related classes in a single file is completely acceptable and improves compile performance. Also, having everything in a single directory makes opening files much simpler at the CLI.

**Comment by driverpt** [comment]
Personally, I think that Code Maintainability is better than save a couple of milliseconds in build time.

Will a PR for this be accepted?

**Comment by lewri** [comment]
I'm personally not against organised structures for files and directories etc.
Whether the PR is accepted is more down to @TheCycoONE  :)

**Comment by TheCycoONE** [comment]
I won't accept a PR that puts every class in it's own file, but I would accept one that splits th_gfx in a simpler way e.g.merges all the code in th_gfx_sdl with the files / classes they belong in and then splits th_gfx by functionality or major classes.

Splitting bitmap and freetype font IMO doesn't help with maintainability, but I welcome opposing views.

Putting all the th_lua_* classes in their own directory (and possibly namespace) makes some sense. I've considered doing that before.

I wouldn't split any of the current classes out of th_map.

Not sure what else you were thinking?

**Comment by driverpt** [comment]
> Splitting bitmap and freetype font IMO doesn't help with maintainability, but I welcome opposing views.

I disagree. When i'm programming in my favourite IDE (VS Code or IntelliJ IDEA), i like to search by File. Having 1 class -> 2 files (.h and .cpp) would make the code more maintainable. Currently i'm constantly doing a search all to find the classes i want and it's a mess and exhausting. But this is my view, and i'm very picky with code maintainability.

**Comment by lewri** [comment]
There's probably an area of compromise here. For keeping neatness in general I usually find @Alberth289346 does give good insight. And as pointed out below by himself:
>I think the main reason for messiness is lack of a description of naming convention of the files. 

I don't touch the C++ of things so from most points of view I can't actually contribute much there. Going back to the wiki page though it looks like the C++ standard is to be followed at most for CorsixTH. I'm not sure what if anything it says about this topic. But if there is some local changes to C++ conventions then make sure the wiki gets it documented too I guess.

**Comment by TheCycoONE** [comment]
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rnr-lots-of-files

**Comment by driverpt** [comment]
I bet the guy who wrote that thinks Unix is user-friendly.

**Comment by Alberth289346** [comment]
I don't want to worry you, but your Apple computer is also a Unix system.
